internal/util/logging.go

```diff
@@ -4,75 +4,105 @@ import (
 	"fmt"
 	"io"
 
-	"github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
-// we currently implement two different loggers and use a middleware called
-// logrusr to translate logrus loggers into go-logrs (used by controller-runtime).
-// The middleware currently squashes loglevels 0-4 together and effectively starts
-// the "info" level logging at 0 (whereas logrus starts that at 4 (range from 0)).
-// Since the middleware at the time of writing made all of this part of the private
-// implementation these options are used for convenience until a time when we unify
-// our logging implementations into one or the other.
-//
-// See: https://github.com/Kong/kubernetes-ingress-controller/issues/1893
+// logr does not use semantic log types. Logs are either informational or error messages with arbitrary verbosity,
+// though errors print regardless of the verbosity (though ensuring this is left up to the implementation--logr
+// interfaces only set the level; they do not filter lines). Only levels above the base info level have meaning.
+// The error and warn levels here are effectively just placeholders.
+
 const (
-	logrusrDiff = 4
+	// ErrorLevel is the logr verbosity level for errors.
+	ErrorLevel = 0
+
+	// WarnLevel is the logr verbosity level for warnings.
+	WarnLevel = 0
 
-	// InfoLevel is the converted logging level from logrus to go-logr for
-	// information level logging. Note that the logrusr middleware technically
-	// flattens all levels prior to this level into this level as well.
-	InfoLevel = int(logrus.InfoLevel) - logrusrDiff
+	// InfoLevel is the logr verbosity level for info logs.
+	InfoLevel = 0
 
-	// DebugLevel is the converted logging level from logrus to go-logr for
-	// debug level logging.
-	DebugLevel = int(logrus.DebugLevel) - logrusrDiff
+	// DebugLevel is the logr verbosity level for debug logs.
+	DebugLevel = 1
 
-	// WarnLevel is the converted logrus level to go-logr for warnings.
-	WarnLevel = int(logrus.WarnLevel) - logrusrDiff
+	// TraceLevel is the logr verbosity level for trace logs.
+	TraceLevel = 2
```

Comment:
Maybe it's worth adjusting the below

https://github.com/Kong/kubernetes-ingress-controller/blob/96d8ed2508ba7478ee3d20c645db091ad66c66c5/internal/manager/config.go#L144

since we're releasing the new major. Do  I understand correctly that it doesn't matter whether I choose `error`, `warn`, `info` the same stuff will be logged? Levels `fatal` and `panic` seems redundant

Reply:
TBD, I wanted to discuss how we actually handle these in sync, since there are a few different options. As-is the levels are squashed, but the config is unchanged to avoid breaking existing config.

Reply:
There is a dedicated issue for making such changes in KIC 3.0.0 milestone
- https://github.com/Kong/kubernetes-ingress-controller/issues/4729